### DISCUSSION TITLE
Vector Extension Recognition in Prisma with Multi-tenant Setup

### DISCUSSION THREAD

**Discussion by emuye** [prisma/prisma#25204] [Q&A]
### Question

I'm working on a multi-tenant application using Prisma with PostgreSQL, where each tenant has its own schema. I'm facing issues with Prisma recognizing the `vector` extension, which is installed in the public schema. Here's the situation:

1. I've added the vector extension to the public schema using a migration:

   ```sql
   -- CreateExtension
   CREATE EXTENSION IF NOT EXISTS "vector" WITH SCHEMA "public";
   ```

2. In my Prisma schema, I initially tried to use `vector(1024)` for the embedding field, but Prisma didn't recognize it. I got the following error:

   ```
   Database error:
   ERROR: type "vector" does not exist
   Position:
     0
     1 -- CreateTable
     2 CREATE TABLE "Embedding" (
     3     "id" SERIAL NOT NULL,
     4     "embedding" vector(1024) NOT NULL,
   ```

3. I've tried setting the search path to include both the tenant-specific schema and the public schema. For example:

   ```sql
   SET search_path TO tenant_schema, public;
   ```

   However, this didn't resolve the issue. Prisma still couldn't recognize the `vector` type.

4. The only way I could get Prisma to even acknowledge that the extension was installed was by explicitly referencing the public schema in the Prisma model:

   ```prisma
   model Embedding {
     id              Int          @id @default(autoincrement())
     embedding       Unsupported("public.vector(1024)")
   }
   ```

   This allowed Prisma to recognize the type, but it feels like a workaround rather than a proper solution.

Why isn't Prisma recognizing the vector extension even though it's installed in the public schema and the search path includes public?

Is there a way to make Prisma aware of extensions in the public schema when working with multiple tenant-specific schemas without having to explicitly reference the public schema in every model?

**Comment by ludralph** [comment]
Hi @emuye 👋 

Can you please take a look at this [issue](https://github.com/prisma/prisma/issues/18442#issuecomment-1518982987) comment? It addresses your exact usecase. Prisma doesn't yet have support for Vectors hence this open [feature request](https://github.com/prisma/prisma/issues/18442).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
